packages/ubuntu_desktop_installer/lib/utils/version_extractor.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+class VersionExtractor {
+  static VersionExtractor? _instance;
+
+  static VersionExtractor get instance {
+    if (_instance == null) {
+      _instance = VersionExtractor._();
+    }
+    return _instance!;
+  }
+
+  VersionExtractor._();
+
+  String loadedVersion = '';
```

Comment:
What about a static `_cachedProductInfo` attribute, instead of making the class a singleton?